salt/metalk8s/addons/logging/loki/deployed/init.sls

```diff
@@ -33,3 +34,98 @@ Delete the old Loki StatefulSet:
       - sls: metalk8s.addons.logging.loki.deployed.chart
 
 {%- endif %}
+
+{#- Delete the Loki pre-upgrade Job since we will deploy the CronJob #}
+Ensure Loki pre-upgrade Workaround Job is deleted:
+  metalk8s_kubernetes.object_absent:
+    - apiVersion: batch/v1
+    - kind: Job
+    - name: loki-cleaner-wa-pre-upgrade
+    - namespace: metalk8s-logging
+    - wait:
+        attempts: 30
+        sleep: 10
+    - require:
+      - sls: metalk8s.addons.logging.loki.deployed.chart
+
+{#- Due to a bug in Loki that prevent deletion of the old chunks,
+    Add a CronJob to delete old chunks waiting for a proper fix #}
+{%- from "metalk8s/repo/macro.sls" import build_image_name with context %}
+{%- set loki_defaults = salt.slsutil.renderer('salt://metalk8s/addons/logging/loki/config/loki.yaml', saltenv=saltenv) %}
+{%- set loki = salt.metalk8s_service_configuration.get_service_conf('metalk8s-logging', 'metalk8s-loki-config', loki_defaults) %}
+
+{#- Let's consider we only support retention_period and index period in hours for the moment #}
+{%- if loki.spec.config.table_manager.retention_period[-1] != "h" %}
+
+Invalid Loki table manager retention_period, should be in hours:
+  test.fail_without_changes: []
+
+{#- NOTE: Consider the last index is the one currently used #}
+{%- elif loki.spec.config.schema_config.configs[-1].index.period[-1] != "h" %}
```

Comment:
ideally we should look for the last config entry using boltdb, but likely ok for now (to. enable us use whatever settings we want for boltdb-shipper which will be the new latest entry)

Reply:
WDYM ? We already use the latest entry of the array.
Or do you mean we should check for the `from` date and take the last one ? (I agree but it's a bit of work just for a temporary workaround)

Reply:
Done